server/core/src/https/views/admin/groups.rs

```diff
@@ -0,0 +1,617 @@
+    Ok(if is_htmx {
+        (push_url, HtmlTemplate(groups_partial)).into_response()
+    } else {
+        (
+            push_url,
+            HtmlTemplate(GroupsView {
+                access_info: AccessInfo::new(),
+                partial: groups_partial,
+            }),
+        )
+            .into_response()
+    })
+}
+
+pub(crate) async fn view_groups_unlock_get(
+    State(state): State<ServerState>,
+    Extension(kopid): Extension<KOpId>,
+    headers: HeaderMap,
+    VerifiedClientInformation(client_auth_info): VerifiedClientInformation,
+    DomainInfo(domain_info): DomainInfo,
+    jar: CookieJar,
+) -> axum::response::Result<Response> {
+    let referrer = match headers.get(header::REFERER) {
+        Some(header_value) => header_value.to_str().map_err(|x| {
+            warn!("referer header couldn't be converted to string: {x}");
+            HtmxError::OperationError(kopid.eventid, OperationError::InvalidRequestState)
+        })?,
+        None => "/ui/admin/groups",
+    };
+    login::view_reauth_get(state, client_auth_info, kopid, jar, referrer, domain_info).await
+}
+
+async fn get_can_rw(
+    state: &ServerState,
+    kopid: &KOpId,
+    client_auth_info: &ClientAuthInfo,
+) -> Result<bool, ErrorResponse> {
+    let uat: UserAuthToken = state
+        .qe_r_ref
+        .handle_whoami_uat(client_auth_info.clone(), kopid.eventid)
+        .map_err(|op_err| HtmxError::new(&kopid, op_err))
+        .await?;
+
+    let time = time::OffsetDateTime::now_utc() + time::Duration::new(60, 0);
+
+    let can_rw = uat.purpose_readwrite_active(time);
+    Ok(can_rw)
+}
+
+async fn get_group_info(
+    uuid: Uuid,
+    state: ServerState,
+    kopid: &KOpId,
+    client_auth_info: ClientAuthInfo,
+) -> Result<GroupInfo, ErrorResponse> {
+    let filter = filter_all!(f_and!([
+        f_eq(Attribute::Class, EntryClass::Group.into()),
+        f_id(uuid.to_string().as_str())
+    ]));
+    let base: Vec<Entry> = state
+        .qe_r_ref
+        .handle_internalsearch(client_auth_info.clone(), filter, None, kopid.eventid)
+        .map_err(|op_err| HtmxError::new(&kopid, op_err))
+        .await?;
+
+    let first = base.first().ok_or(HtmxError::OperationError(
+        kopid.eventid,
+        OperationError::NoMatchingEntries,
+    ))?;
+    Ok(entry_into_groupinfo(first))
+}
+async fn get_groups_info(
+    state: ServerState,
+    kopid: &KOpId,
+    client_auth_info: ClientAuthInfo,
+) -> Result<Vec<GroupInfo>, ErrorResponse> {
+    let filter = filter_all!(f_and!([f_eq(Attribute::Class, EntryClass::Group.into())]));
+    let base: Vec<Entry> = state
+        .qe_r_ref
+        .handle_internalsearch(client_auth_info.clone(), filter, None, kopid.eventid)
+        .map_err(|op_err| HtmxError::new(&kopid, op_err))
+        .await?;
+
+    let mut groups: Vec<_> = base
+        .iter()
+        .map(|entry: &Entry| entry_into_groupinfo(entry))
+        .collect();
+    groups.sort_by_key(|gi| gi.uuid.clone());
```

Comment:
This is going to be super costly, I think we don't sort here - we'll need an open issue to have sorting done in the server backend where it can be done muche more effectively.